### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by endoffile78**

### CODE REVIEW

File: ratox.c [hunk 1]
```diff
@@ -403,6 +413,7 @@ cbcallstart(ToxAV *av, uint32_t cnum, bool audio, bool video, void *udata)
 	logmsg(": %s : Audio > Started\n", f->name);
 }
 
+/*
```
Comment (Diadlo) [inline ratox.c]: You can use `#if 0` to avoid problem with nested comments

File: ratox.c [hunk 2]
```diff
@@ -164,6 +164,14 @@ struct request {
 	TAILQ_ENTRY(request) entry;
 };
 
+#ifdef __GNUC__
```
Comment (Diadlo) [inline ratox.c]: What about using `#define UNUSED(x) (void)(x)` and use this macros in function body?

Comment (GrayHatter) [inline ratox.c]: this is why I like the current way better http://stackoverflow.com/a/12891181

File: ratox.c [hunk 3]
```diff
@@ -1782,7 +1796,7 @@ loop(void)
 			if (c != '0' && c != '1')
 				continue;
 			r = tox_friend_add_norequest(tox, req->id, 0);
-			if (r == UINT32_MAX) {
+			if ((unsigned int)r == UINT32_MAX) {
```
Comment (GrayHatter) [inline ratox.c]: just make r unsigned

File: ratox.c [hunk 4]
```diff
@@ -1330,7 +1342,7 @@ friendcreate(uint32_t frnum)
 	r = tox_friend_get_status(tox, frnum, &err);
 	if (r < 0) {
 		weprintf(": %s : State : Failed to get\n", f->name);
-	} else if (r >= LEN(ustate)) {
+	} else if ((unsigned int)r >= LEN(ustate)) {
```
Comment (GrayHatter) [inline ratox.c]: it's a bad idea to typecast when working with a macro, either or should be rewritten.

File: ratox.c [hunk 5]
```diff
@@ -1166,7 +1178,7 @@ toxinit(void)
 		tcp = 1;
 		toxopt.udp_enabled = !tcp;
 		logmsg("Net > Forcing TCP mode\n");
-		snprintf(toxopt.proxy_host, sizeof(toxopt.proxy_host),
+		snprintf((char *)toxopt.proxy_host, sizeof(toxopt.proxy_host),
```
Comment (GrayHatter) [inline ratox.c]: can we make this non-global.

File: ratox.c (hunk 2 above)
Comment (Diadlo) [inline ratox.c]: @GrayHatter Sorry, from this post I don't understand, why you think, that this way better

**Comment by GrayHatter** [comment]
>

I prefer this because you get an error if you try use bar in the code
anywhere so you can't leave the attribute in by mistake.

**Comment by pranomostro** [comment]
Hello @endoffile78 @GrayHatter @Diadlo,

thanks for working on this project. At the moment, I have little time,
so it's unterstandable that you forked the code, but I will work on it
more in November.

Greetings

File: ratox.c [hunk 6]
```diff
@@ -1112,13 +1124,13 @@ localinit(void)
 
 	/* Dump current name */
 	tox_self_get_name(tox, name);
-	r = tox_self_get_name_size(tox);
-	if (r == 0) {
+	ret = tox_self_get_name_size(tox);
```
Comment (Diadlo) [inline ratox.c]: I think `size` or `name_size` name will be better

File: ratox.c [hunk 7]
```diff
@@ -1128,16 +1140,16 @@ localinit(void)
 	dprintf(gslots[STATUS].fd[OUT], "%s\n", status);
 
 	/* Dump user state */
-	r = tox_self_get_status(tox);
+	ret = tox_self_get_status(tox);
```
Comment (Diadlo) [inline ratox.c]: I think `status` or `self_status` name will be better

**Comment by GrayHatter** [comment]
@iphydf merge please

@Diadlo I agree, but that's out of scope for this pull (IMO)

**Comment by iphydf** [comment]
Please rebase on master.

**Pull request merged**